src/graph/plugin_host.rs

```diff
@@ -176,7 +228,9 @@ pub(crate) struct PluginInstanceHost {
     pub num_audio_in_channels: usize,
     pub num_audio_out_channels: usize,
 
-    main_thread: Option<Box<dyn PluginMainThread>>,
+    // TODO: main thread shouldn't be optional
```

Comment:
That makes sense. The only reason this is optional is because of the "graph in/graph out" nodes. I suppose a way around that is to just have an empty `PluginMainThread` for those nodes.

Reply:
Yes, I only put a TODO since it's not really necessary for this PR, just a small pain point I noticed.

My initial intuition would be to make the in/out nodes either their own `Task` type or something else entirerly, since they're not plugins at all and require much, much less functionality. We can figure out if/how that's doable at a later time though. :slightly_smiling_face: